### PR TITLE
Export plot

### LINKED ISSUE #30: Create a modal for exporting plot
**Is your feature request related to a problem? Please describe.**
Currently, the save button is very simple and limited. 

**Describe the solution you'd like**
We need a modal that will allow the user to pick where they want to save the plot, what file/export type they want to save it as, and possibly what dimensions they want for the plot. The modal
should pop up when user clicks the save/export button. After the user enters their choices, the current canvas would then be saved.

### PR DESCRIPTION
**Pull request by William-Hill**
Resolves #30

### CODE REVIEW

File: src/cell_utils.ts [hunk 1]
```diff
@@ -29,10 +29,14 @@ namespace cell_utils {
       return null;
     }
     let out = cell.outputs.toJSON().pop();
+    console.log("out:", out)
```
Comment (downiec) [inline src/cell_utils.ts]: don't need this console.log anymore, delete

File: src/components/VCSMenu.tsx [hunk 2]
```diff
@@ -337,20 +437,112 @@ export class VCSMenu extends React.Component<VCSMenuProps, VCSMenuState> {
...
+    else{
+      capture = 0
+    }
+    console.log("capture:", capture)
+
+    console.log("height in save:", this.state.height)
+    console.log("width in save:", this.state.width)
+    console.log("plotUnits in save:", this.state.plotUnits)
+
+    if(fileFormat === "png"){
```
Comment (downiec) [inline src/components/VCSMenu.tsx]: Change to: if(fileFormat == "png"), to match the others. Unless there's a reason for it to need '===' ?

**Review by downiec — APPROVED** [review]

File: src/cell_utils.ts (hunk 1 above)
Comment (William-Hill) [reply src/cell_utils.ts]: I removed this and other logging statements with a new commit.

File: src/components/VCSMenu.tsx (hunk 2 above)
Comment (William-Hill) [reply src/components/VCSMenu.tsx]: I used triple equals for strict equivalence checking

**Comment by sterlingbaldwin** [comment]
This is a pretty significant addition to the VCSMenu as far as lines of code goes. I think the modal should be broken out into its own component.

**Comment by downiec** [comment]
I agree, it should be separated into a separate component.

**Review by downiec — CHANGES_REQUESTED** [review]
As pointed out by Sterling, once the code can be made into its own component, I think it will be good to go

**Comment by William-Hill** [comment]
Pushed comments to break the modal into its own component @downiec @sterlingbaldwin

File: src/components/VCSMenu.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Dependencies
 import * as React from "react";
-import { Button, Card, CardBody } from "reactstrap";
-
+import { Alert, Button, ButtonGroup, Card, CardBody, Collapse, CustomInput, Modal, ModalHeader, ModalBody, ModalFooter, Form, FormGroup, Input, Label } from "reactstrap";
```
Comment (downiec) [inline src/components/VCSMenu.tsx]: Its good to remove any unnecessary imports and leave them only in the file they're needed. For now, I'll remove them in the next PR after this one is merged.

**Review by downiec — APPROVED** [review]

**Review by downiec — CHANGES_REQUESTED** [review]
Just tested it, and I noticed that every time I click the toggle button to enter my own custom dimensions, the dimensions get injected into the notebook. I think it would be better if the injection is done after the user finalizes their decisions and clicks 'export', otherwise the notebook can quickly get spammed with injected code. Also I keep getting an error: 'Code resulted in errors. Error name: Exception. Message: Nothing to dump aborting.' so we need to get that fixed before merging.

**Review by downiec — APPROVED** [review]

File: src/components/ExportPlotModal.tsx [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+                  name="height"
+                  placeholder="Height"
+                  value={this.state.height}
+                  onChange={e => this.setState({ height: e.target.value })}
+                />
+              </Collapse>
+            </div>
+          </div>
+          <br />
+          {/* <CustomInput type="switch" id="exampleCustomSwitch" name="customSwitch" label="Capture Provenance" checked={this.state.captureProvenance} onChange={this.toggleCaptureProvenance} /> */}
```
Comment (sterlingbaldwin) [inline src/components/ExportPlotModal.tsx]: Commented out code should be removed before its merged.

File: src/components/VCSMenu.tsx [hunk 5]
```diff
@@ -325,20 +408,23 @@ export class VCSMenu extends React.Component<VCSMenuProps, VCSMenuState> {
...
         plotString = "canvas.plot(";
-      }
-      else {
-          plotString = "canvas.clear()\ncanvas.plot(";
+      } else {
+        plotString = "canvas.clear()\ncanvas.plot(";
       }
 
-      console.log("plotString:", plotString)
+      console.log("plotString:", plotString);
```
Comment (sterlingbaldwin) [inline src/components/VCSMenu.tsx]: Debug console logs should be removed before merge.

**Review by sterlingbaldwin — CHANGES_REQUESTED** [review]
Just a couple minor issues, nothing serious.

**Comment by William-Hill** [comment]
Just removed the comment and removed the provenance capture. I'll create a separate PR for the provenance capture after the backend part is fixed @sterlingbaldwin

**Comment by sterlingbaldwin** [comment]
I just merged the help_links PR, which includes all these changes so Im closing this.

**Pull request closed without merging**